src/query/storage/m3/multi_fetch_result.go

```diff
@@ -89,19 +117,17 @@ func (r *multiResult) FinalResult() (encoding.SeriesIterators, error) {
 	}
 
 	// otherwise have to create a new seriesiters
-	iter := r.pools.MutableSeriesIterators().Get(numSeries)
-	iter.Reset(numSeries)
+	numSeries := len(r.dedupeMap)
+	r.finalResult = r.pools.MutableSeriesIterators().Get(numSeries)
+	r.finalResult.Reset(numSeries)
 
 	i := 0
 	for _, res := range r.dedupeMap {
-		if len(res) != 1 {
-			return nil, fmt.Errorf("internal error during result dedupe, expected 1 id, observed: %d", len(res))
-		}
-		iter.SetAt(i, res[0].iter)
+		r.finalResult.SetAt(i, res.iter)
 		i++
 	}
 
-	return iter, nil
+	return r.finalResult, nil
 }
 
 func (r *multiResult) Add(
```

Comment:
super nit: maybe call the iterators arg newIterators, I got a little lost between `seenIters` and `iterators` the first time

Reply:
Sure thing.